lxd/identities.go

```diff
@@ -1274,7 +1274,7 @@ func updateIdentity(authenticationMethod string) func(d *Daemon, r *http.Request
 			return response.BadRequest(fmt.Errorf("Failed to unmarshal request body: %w", err))
 		}
 
-		if id.Type != api.IdentityTypeCertificateClient && identityPut.TLSCertificate != "" {
+		if id.Type != api.IdentityTypeCertificateClient && id.Type != api.IdentityTypeCertificateClusterLink && identityPut.TLSCertificate != "" {
```

Comment:
@kadinsayani please follow up with defining a list of this concept of identity types and then de-dupe this file with the repeat logic surrounding types.